File: code/mary/ATM.py
#ATM.py

class Account:
    def __init__(self, balance = 0.0):
        self.balance = balance
        self.history = []

    def check_balance(self):
        return self.balance

    def deposit(self, amount):
        self.balance = self.balance + amount
        self.history.append(f"User deposited ${amount}.")
        return self.balance

    def check_withdrawal(self, amount):
        if self.balance>self.balance:
            return True
        else:
            return False

    def withdraw(self, amount):
        self.balance = self.balance - amount
        self.history.append(f"User withdrew ${amount}.")
        return self.balance

    def print_transactions(self):
        return self.history
        
acc = Account()
def repl(acc):
    while True:
        user_input = input(f"What would you like to do: deposit, withdraw, check balance, check history or exit? " )
        if user_input == "deposit":
            user_amt = int(input("How much would you like to deposit? "))
            (acc.deposit(user_amt))
        elif user_input == "withdraw":
            user_amt = int(input("How much would you like to withdraw? "))
            (acc.withdraw(user_amt))
        elif user_input == "check balance":
            print(acc.check_balance())
        elif user_input == "check history":
            print(acc.print_transactions())
        elif user_input == "exit":
            break
repl(acc)

File: code/tate/Labs/JavaScript/lab1-3/atm.js
// Lab 25 : ATM
let atmBtnDeposit = document.querySelector('#atmBtnDeposit');
let atmBtnWithdraw = document.querySelector('#atmBtnWithdraw');
let atmBalance = document.querySelector('#atmBalance');
let atmTransactions = document.querySelector('#atmTransactions');

class ATM {
  constructor(balance=0) {
    this.balance = balance;
    this.transactions = [];
  }
  checkBalance () {
    return this.balance;
  }
  deposit (amount) {
    this.balance += amount;
    this.transactions.push('User deposited: ' + `${amount}`);
    return this.balance;
  }
  checkWithdraw (amount) {
    if (amount <= this.balance){
      return true;
    } else {
      return false;
    }
  }
  withdraw (amount) {
    if (this.checkWithdraw(amount)){
      this.balance -= amount;
      this.transactions.push('User withdrew: ' + `${amount}`);

    } else {
      alert('Insufficient funds');
    }
    return this.balance;
  }
  printTransactions (){
    return this.transactions.join('\n');
  }
}
function atmInit(){
  atmBalance.innerText = `Account Balance : ${atm1.checkBalance()}`;
}

function atmTransactionsFunc () {
  atmTransactions.innerText = `${atm1.printTransactions()}`;
}

function atmFuncDeposit() {
  let depositAmount = parseInt(document.querySelector('#atmDeposit').value);
  atm1.deposit(depositAmount);
  atmBalance.innerText = `Account Balance : ${atm1.checkBalance()}`;
  atmTransactionsFunc();
}

function atmFuncWithdraw() {
  let withdrawAmount = parseInt(document.querySelector('#atmWithdraw').value);
  atm1.withdraw(withdrawAmount);
  atmBalance.innerText = `Account Balance : ${atm1.checkBalance()}`;
  atmTransactionsFunc();
}
var atm1 = new ATM(0);
atmInit();

atmBtnDeposit.addEventListener('click', atmFuncDeposit);
atmBtnWithdraw.addEventListener('click', atmFuncWithdraw);
